### ISSUE TITLE
Add symbol name for narrow non-breaking space

### ISSUE THREAD

**Issue by t-rapp** [typst/typst#3197]
### Description

It would be nice to add a symbol name for the narrow variant of the non-breaking space character (U+202F).

Suggested symbol name would be `sym.space.nobreak.narrow`.

### Use Case

The narrow variant of the non-breaking space character is useful when writing text containing numbers and units. Placing both parts closer together may improve the visual appearance.

Example:

 ```
#show "_": sym.space.nobreak.narrow
Weight: 3_456_g, Size: 54_cm

// versus

Weight: 3~456~g, Size: 54~cm
```

**Comment by t-rapp** [comment]
From a quick test it seems enough to add an entry for that Unicode character to `crates/typst/src/symbols/sym.rs`.

Would it be needed to add a test case for the non-line-breaking property?

**Comment by t-rapp** [comment]
Have included some tests in pull request #3217.
